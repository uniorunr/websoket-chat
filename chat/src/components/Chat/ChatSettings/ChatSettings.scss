@import "../../../theme/colors";

.chat-settings {
  position: absolute;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  width: 95%;
  height: 95%;
  background: $white-color;
  border: 2px solid $black-color;
  border-radius: 5px;
  z-index: 1;

  .chat-settings__background {
    width: 90%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    margin: 0 auto;

    h3 {
      margin: 30px 0 10px;
      font-size: 1.4rem;
    }

    ul {
      display: flex;
      justify-content: space-around;
      width: 100%;
      list-style: none;
      padding: 0;
      margin: 0;
      flex-wrap: wrap;

      li {
        margin: 10px;
        cursor: pointer;

        img {
          border: 2px solid transparent;
        }

        .active-img {
          border: 2px solid $black-color;
        }
      }
    }
  }

  .submit {
    width: 50%;
    background: $black-color;
    color: $white-color;
    text-transform: uppercase;
    padding: 10px 20px;
    outline: none;
    cursor: pointer;
    border: 2px solid $black-color;
    border-radius: 5px;
    margin: 30px;
  }

  .submit:hover {
    background: $white-color;
    color: $black-color;
  }

  .close {
    position: absolute;
    top: -2px;
    right: -2px;
    background: $white-color;
    border: 2px solid $black-color;
    border-radius: 0 5px 0 0;
    font-size: 1.4rem;
    font-weight: 700;
    cursor: pointer;
    outline: none;
  }
}
