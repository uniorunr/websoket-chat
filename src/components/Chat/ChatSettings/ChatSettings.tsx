import React, { Component, MouseEvent } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators, Dispatch } from 'redux';
import * as actions from '../../../actions/actions';
import { StoreState } from '../../../types';
import HeartsPattern from '../../../assets/background/hearts.jpg';
import LeafsPattern from '../../../assets/background/leafs.jpg';
import HexalPattern from '../../../assets/background/hexals.jpg';
import CrossesPattern from '../../../assets/background/crosses.jpg';
import TrianglesPattern from '../../../assets/background/triangles.jpg';
import ReactPattern from '../../../assets/background/papyrus.jpg';
import MoroccoPattern from '../../../assets/background/morocco.jpg';
import './ChatSettings.scss';

const chatBackgrounds = [
  { id: 'hearts', path: HeartsPattern },
  { id: 'leafs', path: LeafsPattern },
  { id: 'hexals', path: HexalPattern },
  { id: 'crosses', path: CrossesPattern },
  { id: 'triangles', path: TrianglesPattern },
  { id: 'papyrus', path: ReactPattern },
  { id: 'morocco', path: MoroccoPattern }
];

class ChatSettings extends Component<ChatSettingsProps> {
  state = {
    selectedBackground: ''
  };

  componentDidMount(): void {
    const { activeBackgroundId } = this.props;
    this.setState({
      selectedBackground: activeBackgroundId
    });
  }

  handleClickOnBackground = (event: MouseEvent<HTMLElement>) => {
    const { selectedBackground } = this.state;

    if (selectedBackground !== event.currentTarget.dataset.id) {
      this.setState({
        selectedBackground: event.currentTarget.dataset.id
      });
    } else {
      this.setState({
        selectedBackground: ''
      });
    }
  };

  handleSubmit = () => {
    const { updateBackgroundId, showSettingsModalWindow } = this.props;
    const { selectedBackground } = this.state;
    updateBackgroundId(selectedBackground);
    showSettingsModalWindow(false);
    localStorage.setItem('chatBackground', selectedBackground);
  };

  render() {
    const { showSettingsModalWindow } = this.props;
    const { selectedBackground } = this.state;

    return (
      <section className="chat-settings">
        <div className="chat-settings__background">
          <h3>Chat background</h3>
          <ul>
            {chatBackgrounds.map(background => (
              <li
                key={background.id}
                data-id={background.id}
                onClick={this.handleClickOnBackground}
              >
                <img
                  src={background.path}
                  alt="Background pattern"
                  width={75}
                  className={
                    background.id === selectedBackground ? 'active-img' : ''
                  }
                />
              </li>
            ))}
          </ul>
        </div>
        <button type="button" className="submit" onClick={this.handleSubmit}>
          submit
        </button>
        <button
          type="button"
          className="close"
          onClick={() => showSettingsModalWindow(false)}
        >
          x
        </button>
      </section>
    );
  }
}

interface ChatSettingsProps {
  showSettingsModalWindow: Function;
  updateBackgroundId: Function;
  activeBackgroundId: string;
}

const mapStateToProps = (state: StoreState) => ({
  activeBackgroundId: state.settings.activeBackgroundId
});

const mapDispatchToProps = (dispatch: Dispatch) => {
  const { showSettingsModalWindow, updateBackgroundId } = bindActionCreators(
    actions,
    dispatch
  );
  return {
    showSettingsModalWindow: (status: boolean) => {
      showSettingsModalWindow(status);
    },
    updateBackgroundId: (id: string) => {
      updateBackgroundId(id);
    }
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ChatSettings);
