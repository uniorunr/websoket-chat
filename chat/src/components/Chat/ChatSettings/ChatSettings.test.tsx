import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import store from '../../../store/mainStore';
import ChatSettings from './ChatSettings';

it('ChatSettings component renders correctly', () => {
  const app = shallow(
      <Provider store={store}>
        <ChatSettings />
      </Provider>
  );
  expect(app).toMatchSnapshot();
});
