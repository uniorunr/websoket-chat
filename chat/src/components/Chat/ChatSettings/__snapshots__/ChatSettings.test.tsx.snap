// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatSettings component renders correctly 1`] = `ShallowWrapper {}`;
